compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -918,7 +917,40 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                         gen_args.span(),
                         "parenthesized generic arguments cannot be used in associated type constraints"
                     );
-                    // FIXME: try to write a suggestion here
+                    if let Ok(snippet) = self.sess.source_map().span_to_snippet(data.span) {
```

Comment:
Since this whole error reporting got a bit more complex, put everything from the `struct_span_err` to the `emit` into a separate function so that this function becomes easier to read.